src/Reads.cpp

```diff
@@ -37,6 +37,55 @@ void Reads::access(
 }
 
 
+void Reads::renameWithSuffix(const string& suffix) {
+    const string readsDataName = reads.getName();
+    const string readNamesDataName = readNames.getName();
+    const string readMetaDataDataName = readMetaData.getName();
+    const string readRepeatCountsDataName = readRepeatCounts.getName();
+    const string readFlagsDataName = readFlags.fileName;
+
+    reads.rename(readsDataName + suffix);
+    readNames.rename(readNamesDataName + suffix);
+    readMetaData.rename(readMetaDataDataName + suffix);
+    readRepeatCounts.rename(readRepeatCountsDataName + suffix);
+    readFlags.rename(readFlagsDataName + suffix);
+}
+
+
+void Reads::copyDataForReadsLongerThan(const Reads& rhs, uint64_t newMinReadLength) {
+    for(ReadId id = 0; id < rhs.readCount(); id++) {
+        const auto len = rhs.getReadRawSequenceLength(id);
+        if (len > newMinReadLength) {
+            // Copy over stuff.
+            readNames.appendVector(rhs.readNames.begin(id), rhs.readNames.end(id));
+            readMetaData.appendVector(rhs.readMetaData.begin(id), rhs.readMetaData.end(id));
+            reads.append(rhs.reads[id]);
+            const uint64_t j = readRepeatCounts.size();
+            readRepeatCounts.appendVector(rhs.readRepeatCounts.size(id));
```

Comment:
Could this also use a pattern similar to the above calls to `appendVector`, without using a `copy`? Any special reason why this one is done differently?

Reply:
May be. I copy-pasted this code without thinking about potentially rewriting it. This is similar to the code we use to transfer data from per-thread datastructures into the main ones.